File: org.summer.view.widget/src/org/summer/view/widget/controls/ItemContainerTemplate.java
package org.summer.view.widget.controls;

import org.summer.view.widget.DataTemplate;
/// <summary>
    ///   The template that produces a container for an ItemsControl
    /// </summary>
//    [DictionaryKeyProperty("ItemContainerTemplateKey")] 
    public class ItemContainerTemplate extends DataTemplate
    { 
 
        /// <summary>
        ///     The key that will be used if the ItemContainerTemplate is added to a 
        ///     ResourceDictionary in Xaml without a specified Key (x:Key).
        /// </summary>
        public Object ItemContainerTemplateKey
        { 
            get
            { 
                return (DataType != null) ? new ItemContainerTemplateKey(DataType) : null; 
            }
        } 
    }

File: summer/WebContent/js/system/windows/controls/ItemContainerTemplate.js
/**
 * ItemContainerTemplate
 */

define(["dojo/_base/declare", "system/Type", "windows/DataTemplate", "controls/ItemContainerTemplateKey"], 
		function(declare, Type, DataTemplate, ItemContainerTemplateKey){
	var ItemContainerTemplate = declare("ItemContainerTemplate", DataTemplate,{
		constructor:function(){
		}
	});
	
	Object.defineProperties(ItemContainerTemplate.prototype,{
        /// <summary>
        ///     The key that will be used if the ItemContainerTemplate is added to a 
        ///     ResourceDictionary in Xaml without a specified Key (x:Key).
        /// </summary>
//        public object 
		ItemContainerTemplateKey:
        { 
            get:function()
            { 
                return (this.DataType != null) ? new ItemContainerTemplateKey(this.DataType) : null; 
            }
        }   
	});
	
	Object.defineProperties(ItemContainerTemplate,{
		  
	});
	
	ItemContainerTemplate.Type = new Type("ItemContainerTemplate", ItemContainerTemplate, [DataTemplate.Type]);
	return ItemContainerTemplate;
});

